### ISSUE TITLE
Cms block cache key does not contain the store id

### ISSUE THREAD

**Issue by tzyganu** [magento/magento2#22299]
### Preconditions (*)
1. Magento version 2.2.6
2. create one block that contains a `{{trans}}` directive. Or 2 blocks with the same identifier but each one available on a different store view
3. Block cache must be enabled.

### Steps to reproduce (*)
1. Add the block in the layout and specify a cache lifetime

```
            <block class="Magento\Cms\Block\Block" name="some-name-here">
                <arguments>
                    <argument name="block_id" xsi:type="string">block-identifier-here</argument>
                    <argument name="cache_lifetime" xsi:type="number">84600</argument>
                </arguments>
            </block>
```

### Expected result (*)

1. Loading the pages that contain this block on 2 different store views will show the proper content of the block for each website

### Actual result (*)

1. Opening the page that contains the block on the second store view shows the cached version of the block from the first store view.

### Additional info:
This happens because the cache key of the cms block is determined by `Magento\Framework\View\Element\AbstractBlock::getCacheKey` which calls `getCacheKeyInfo` from the same class which returns an array with the name of the block in the layout.  

###  Possible solution:
Adding the store id to the cache key or to the array returned by `getCacheKeyInfo`.

**Comment by ghost** [comment]
HI @tzyganu thank you for you report, Would you like to create Pull Request with this ?

**Comment by tzyganu** [comment]
@engcom-backlog-nazar If you really want me to do it, sure. Unless you want to take over. if you do, I won't oppose.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99156`, `MAGETWO-99157` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by dipti2jcommerce** [comment]
@magento-engcom-team give me 2.3-develop instance - upcoming 2.3.x release

**Comment by magento-engcom-team** [comment]
Hi @dipti2jcommerce. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @tzyganu. Thank you for your report. 
The issue has been fixed in magento/magento2#22302 by @tzyganu in 2.3-develop branch
Related commit(s):
 - [5c84c1ad6aecbf6209d04bcaa160c49f356344d4](https://github.com/magento/magento2/commit/5c84c1ad6aecbf6209d04bcaa160c49f356344d4)
 - [97bdecc78dde3a0c6d8eeaad98ecdfe50e8575be](https://github.com/magento/magento2/commit/97bdecc78dde3a0c6d8eeaad98ecdfe50e8575be)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @tzyganu. Thank you for your report. 
The issue has been fixed in magento/magento2#22534 by @amol2jcommerce in 2.2-develop branch
Related commit(s):
 - [d707c6b78d508ccf2d5d519fffa2b32a5ee997a3](https://github.com/magento/magento2/commit/d707c6b78d508ccf2d5d519fffa2b32a5ee997a3)
 - [c17b2328f8d8d881fbebd0a4c40a88b4ae465784](https://github.com/magento/magento2/commit/c17b2328f8d8d881fbebd0a4c40a88b4ae465784)

The fix will be available with the upcoming 2.2.9 release.
